consensus/src/main/java/org/ethereum/beacon/consensus/SpecHelpers.java

```diff
@@ -1002,12 +1009,15 @@ def eligible(index):
 
   /** Function for hashing objects into a single root utilizing a hash tree structure */
   public Hash32 hash_tree_root(Object object) {
-    return objectHasher.getHash(object);
+    String id = object.getClass().getName() + "@" + Integer.toHexString(object.hashCode());
```

Comment:
I think `System.identityHashCode(object)` would be more suitable here

Reply:
outdate